src/utils/fetcher.tsx

```diff
@@ -0,0 +1,20 @@
+function fetcher<TResponse>(
+  url: string,
+  // `RequestInit` is a type for configuring
+  // a `fetch` request. By default, an empty object.
+  config: RequestInit = {}
+
+  // This function is async, it will return a Promise:
+): Promise<TResponse> {
+  // Inside, we call the `fetch` function with
+  // a URL and config given:
+  return (
+    fetch(url, config)
+      // When got a response call a `json` method on it
+      .then((response) => response.json())
+      // and return the result data.
+      .then((data) => data as TResponse)
+  );
```

Comment:
I always prefer to use `async/await` syntax instead of `.then`. I makes code cleaner. And force you to mark function as async (note that you marked a `fetchPokemons` as async, but not a `fetcher`